### PR TITLE
Correctly format duplicate fields

### PR DESCRIPTION
**Pull request by Gabriel439**
Fixes https://github.com/dhall-lang/dhall-haskell/issues/2032

### CODE REVIEW

File: dhall/tests/format/duplicatesB.dhall [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{ foo.bar = 1
+, foo.baz
+  = {- Comments within values are still preserved-}
+    2
```
Comment (sjakobi) [inline dhall/tests/format/duplicatesB.dhall:4]: This is unrelated to this PR, but makes me realize that maybe the `=` should rather be on the same line as the dotted fields:

```suggestion
, foo.baz =
    {- Comments within values are still preserved-}
    2
```

Having the `=` below the keys is IMHO mostly warranted when the keys are spread over several lines and especially when the last key has a trailing comment, e.g.

```
{ k
  {- bla -}
  = v
}
```

Thoughts? (We can move this discussion to a separate issue though.)

**Review by sjakobi — APPROVED** [review]
LGTM!

File: dhall/tests/format/duplicatesB.dhall (hunk 1 above)
Comment (Gabriel439) [reply dhall/tests/format/duplicatesB.dhall:4]: @sjakobi: It should probably be a separate issue.  I try to keep changes logically separate if possible

Comment (sjakobi) [reply dhall/tests/format/duplicatesB.dhall:4]: I opened https://github.com/dhall-lang/dhall-haskell/issues/2051 to track this.

**Pull request merged**